javascript/app-framework/src/components/DataTableConfig/Parser/Parser.tsx

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ParserOptions } from '@datashaper/workflow'
+import { Expando } from '@essex/components'
+import {
+	Checkbox,
+	ChoiceGroup,
+	Position,
+	SpinButton,
+	TextField,
+} from '@fluentui/react'
+import { memo } from 'react'
+
+import { useUnobservedProperty } from '../../../hooks/index.js'
+import { Delimiter } from './Delimiter.js'
+import { Headers } from './Headers.js'
+import {
+	Container,
+	expandoStyles,
+	FieldContainer,
+	FlexContainer,
+} from './Parser.styles.js'
+import type { ParserProps } from './Parser.types.js'
+
+const lineTerminatorOptions = [
+	{ key: '\r', text: '\\r' },
+	{ key: '\n', text: '\\n' },
+	{ key: '\r\n', text: '\\r\\n' },
+]
+
+export const Parser: React.FC<ParserProps> = memo(function Parser({ parser }) {
+	const skipRows = useUnobservedProperty<ParserOptions, number>(
+		parser,
+		'skipRows',
```

Comment:
Is ParserOptions the main use case for this hook? Maybe we should just make these properties observables properly.